contract_split/models/contract.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2023 Damien Crier - Foodles
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class Contract(models.Model):
+    _inherit = "contract.contract"
+
+    original_contract_ids = fields.Many2many(
+        comodel_name="contract.contract",
+        relation="contract_split_contract_rel",
+        column1="contract_id",
+        column2="split_contract_id",
+        readonly=True,
+    )
+
+    @api.model
+    def _get_contract_split_name(self, split_wizard):
+        return split_wizard.main_contract_id.name
+
+    @api.model
+    def _get_values_create_split_contract(self, split_wizard):
+        return {
+            "name": self._get_contract_split_name(split_wizard),
+            "partner_id": split_wizard.partner_id.id,
+            "invoice_partner_id": split_wizard.invoice_partner_id.id,
+            "original_contract_ids": [split_wizard.main_contract_id.id],
+            "line_recurrence": True,
+        }
+
+    def _get_default_split_values(self) -> dict:
+        self.ensure_one()
+        return {
+            "main_contract_id": self.id,
+            "partner_id": self.partner_id.id,
+            "invoice_partner_id": self.invoice_partner_id.id,
+            "split_line_ids": [
+                (0, 0, line._get_default_split_line_values())
+                for line in self.contract_line_ids
+            ],
+        }
+
+
+class ContractLine(models.Model):
+    _inherit = "contract.line"
+
+    splitted_from_line_id = fields.Many2one(
+        comodel_name="contract.line",
+        readonly=True,
+    )
+    splitted_from_contract_id = fields.Many2one(
+        comodel_name="contract.contract",
```

Comment:
We could probably use a related field here using `splitted_from_line_id.contract_id` ?

Reply:
ok I realize if tomorrow we implement a move we know that split came from such contract !

Reply:
yes and if the line is completly moved from one contract to another, there is no origin contract line